metabase/metabase
# Add warning message when creating or editing a database connection

### Description

Add `beforeunload` hook when the user is trying to edit or create a database connection.

### How to verify

1. Go to Metabase Admin and try to change the name of a database that you created, or just try to create a database and add some information
2. Try to refresh/close/navigate away from the page, and you should see a warning message from the browser

### Demo

_Upload a demo video or before/after screenshots if sensible or remove the section_

### Checklist



- - -
This change is [<img src="https://reviewable.io/review_button.svg" height="34" align="absmiddle" alt="Reviewable"/>](https://reviewable.io/reviews/metabase/metabase/30386)
